File: cms/src/main/java/com/tqp/cms/service/PrescriptionService.java
package com.tqp.cms.service;

import com.tqp.cms.dto.response.PrescriptionResponse;
import com.tqp.cms.dto.response.PageResponse;
import java.time.LocalDate;
import java.util.UUID;

public interface PrescriptionService {
    PrescriptionResponse getPrescriptionById(UUID prescriptionId);

    PrescriptionResponse getPrescriptionByMedicalRecordId(UUID medicalRecordId);

    PageResponse<PrescriptionResponse> getMyPrescriptions(LocalDate issuedDate, int page, int size);
}

File: clinic-app/src/services/prescriptionService.js
import { api } from './api.js'

export async function getPrescriptionById(prescriptionId) {
  const { data } = await api.get(`/prescriptions/${prescriptionId}`)
  return data
}

export async function getPrescriptionByMedicalRecordId(medicalRecordId) {
  const { data } = await api.get(`/prescriptions/medical-record/${medicalRecordId}`)
  return data
}

